src/js/trove/parse-pyret.js

```diff
@@ -1387,6 +107,48 @@
       "raises-violates":   function(l){return RUNTIME.getField(ast, "s-op-raises-violates").app(l);},
     }
 
+    function translate(node, fileName) {
+      let constructors = {
+        "makeNode": function(con) {
+          let args = Array.prototype.slice.call(arguments, 1);
+          if (args.length === 0) {
+            return RUNTIME.getField(ast, con);
+          } else {
+            return RUNTIME.getField(ast, con).app.apply(this, args);
+          }
+        },
+        "opLookup": opLookup,
+        "makeLink": function(head, tail) {
```

Comment:
I don't think any of these need to be eta-expanded, do they?

Reply:
makeLink has fixed arity, so it's fine. opLookup was a dictionary that returned either values for binops or (srcloc->value) functions for checkops, which was a little clunky so I'm replacing it with two separate dictionaries for binops and checkops.